recipes/opencv/2.x/conanfile.py

```diff
@@ -0,0 +1,248 @@
+
+        def get_components():
+            components = []
+            if self.options.with_jasper:
+                components.append("jasper::jasper")
+            if self.options.with_png:
+                components.append("libpng::libpng")
+            if self.options.with_jpeg:
+                components.append("libjpeg::libjpeg")
+            if self.options.with_tiff:
+                components.append("libtiff::libtiff")
+            if self.options.with_openexr:
+                components.append("openexr::openexr")
+            if self.options.with_eigen:
+                components.append("eigen::eigen")
+            if self.options.with_tbb:
+                components.append("tbb::tbb")
+            return components
+
+        def eigen():
+            return ["eigen::eigen"] if self.options.with_eigen else []
+
+        def tbb():
+            return ["tbb::tbb"] if self.options.with_tbb else []
+
+        self.cpp_info.filenames["cmake_find_package"] = "OpenCV"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "OpenCV"
+
+        add_components({
+            "core": ["zlib::zlib"] + tbb(),
+            "flann": ["core"] + tbb(),
+            "imgproc": ["core"] + tbb(),
+            "highgui": ["core", "imgproc"] + get_components(),
+            "features2d": ["core", "flann", "imgproc", "highgui"] + tbb(),
+            "calib3d": ["core", "flann", "imgproc", "highgui", "features2d"] + tbb(),
+            "ml": ["core"] + tbb(),
+            "video": ["core", "imgproc"] + tbb(),
+            "legacy": ["core", "flann", "imgproc", "highgui", "features2d", "calib3d", "ml", "video"] + eigen() + tbb(),
+            "objdetect": ["core", "imgproc", "highgui"] + tbb(),
+            "photo": ["core", "imgproc"] + tbb(),
+            "gpu": ["core", "flann", "imgproc", "highgui", "features2d", "calib3d", "ml", "video", "legacy", "objdetect", "photo"] + tbb(),
+            "nonfree": ["core", "flann", "imgproc", "highgui", "features2d", "calib3d", "ml", "video", "legacy", "objdetect", "photo", "gpu"] + tbb(),
+            "contrib": ["core", "flann", "imgproc", "highgui", "features2d", "calib3d", "ml", "video", "legacy", "objdetect", "photo", "gpu", "nonfree"] + tbb(),
+            "stitching": ["core", "flann", "imgproc", "highgui", "features2d", "calib3d", "ml", "video", "legacy", "objdetect", "photo", "gpu", "nonfree"] + tbb(),
+            "superres": ["core", "flann", "imgproc", "highgui", "features2d", "calib3d", "ml", "video", "legacy", "objdetect", "photo", "gpu"] + tbb(),
+            "ts": ["core", "flann", "highgui", "features2d", "calib3d", "video"] + tbb(),
+            "videostab": ["core", "flann", "imgproc", "highgui", "features2d", "calib3d", "ml", "video", "legacy", "objdetect", "photo", "gpu"] + tbb()
+        })
+
+        if self.settings.os == "Macos":
+            self.cpp_info.components["highgui"].frameworks = ["Cocoa"]
```

Comment:
On Windows, I see several system libs for `opencv_highgui` target in `OpenCVModules.cmake` (I've disabled all dependencies except zlib):

```cmake
set_target_properties(opencv_highgui PROPERTIES
  INTERFACE_LINK_LIBRARIES "opencv_core;opencv_imgproc;opencv_core;opencv_imgproc;\$<LINK_ONLY:comctl32>;\$<LINK_ONLY:gdi32>;\$<LINK_ONLY:ole32>;\$<LINK_ONLY:setupapi>;\$<LINK_ONLY:ws2_32>;\$<LINK_ONLY:vfw32>;\$<LINK_ONLY:comctl32>;\$<LINK_ONLY:gdi32>;\$<LINK_ONLY:ole32>;\$<LINK_ONLY:setupapi>;\$<LINK_ONLY:ws2_32>;\$<LINK_ONLY:vfw32>"
)
```

Reply:
Done